src/app/core/services/settings.ts

```diff
@@ -151,22 +151,29 @@ export class SettingsService {
     this._usersRefresh$.next();
   }
 
-  get reports(): Observable<Report[]> {
-    if (!this._reports$) {
-      this._reports$ = this._refreshTimer$
-        .pipe(switchMap(() => this._getReports()))
+  reports(scheduleName: string): Observable<Report[]> {
+    if (!this._reports$.get(scheduleName)) {
+      const reports$ = this._refreshTimer$
+        .pipe(switchMap(() => this._getReports(scheduleName)))
         .pipe(shareReplay({refCount: true, bufferSize: 1}));
+      this._reports$.set(scheduleName, reports$);
     }
-    return this._reports$;
+
+    return this._reports$.get(scheduleName);
   }
 
-  private _getReports(): Observable<Report[]> {
+  private _getReports(scheduleName: string): Observable<Report[]> {
     const url = `${this._restRoot}/admin/metering/reports`;
-    return this._httpClient.get<Report[]>(url);
+    return this._httpClient.get<Report[]>(url, {params: {configuration_name: scheduleName}});
   }
 
-  reportDownload(reportName: string): Observable<string> {
-    const url = `${this._restRoot}/admin/metering/reports/${reportName}`;
-    return this._httpClient.get<string>(url);
+  reportDownload(reportPath: string, scheduleName?: string): Observable<string> {
```

Comment:
Since the `scheduleName` is now required to get reports, why is it not required for download? Is the API able to download it anyway without providing the `scheduleName`?

Reply:
Fair point. Technically old reports can still be located at the root level of an object store, but we shouldn't care about them and run migration script to place them in a proper directory (named after `scheduleName`) instead. Let me update the code to make this argument required.